application/src/ext/java/org/opentripplanner/ext/carpooling/util/DirectionalCalculator.java

```diff
@@ -0,0 +1,59 @@
+package org.opentripplanner.ext.carpooling.util;
+
+import org.opentripplanner.framework.geometry.WgsCoordinate;
+
+/**
+ * Calculates bearings and directional relationships between geographic coordinates.
+ * <p>
+ * Uses the Haversine formula for accurate bearing calculations on Earth's surface.
+ * All bearings are in degrees [0, 360) where 0° = North, 90° = East, etc.
+ */
+public class DirectionalCalculator {
+
+  /**
+   * Calculates the initial bearing (forward azimuth) from one point to another.
+   * <p>
+   * Uses the Haversine formula for accurate bearing on Earth's surface.
+   *
+   * @param from Starting point
+   * @param to Ending point
+   * @return Bearing in degrees [0, 360), where 0° is North
+   */
+  public static double calculateBearing(WgsCoordinate from, WgsCoordinate to) {
```

Comment:
I wonder if we already have this somewhere or if some library we use has support for it.